File: backend/app/models/camera.py
import datetime as dt

from sqlalchemy import Column, DateTime, Enum, Float, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship

from app.database import Base
from app.models.enums import CameraStatusEnum, CameraTypeEnum


class Camera(Base):
    __tablename__ = "cameras"

    id = Column(Integer, primary_key=True, index=True)
    store_id = Column(Integer, ForeignKey("stores.id"), nullable=False)
    zone_id = Column(Integer, ForeignKey("store_zones.id"), nullable=True)

    name = Column(String(150), nullable=False)
    camera_type = Column(Enum(CameraTypeEnum), nullable=False, default=CameraTypeEnum.WEBCAM)
    status = Column(Enum(CameraStatusEnum), nullable=False, default=CameraStatusEnum.CONFIGURING)

    stream_url = Column(String(500), nullable=True)  # RTSP / IP camera URL
    resolution_width = Column(Integer, nullable=True)
    resolution_height = Column(Integer, nullable=True)
    fps = Column(Integer, nullable=True)

    # homography / calibration info mapping camera pixels -> floor coordinates (JSON string)
    calibration_data = Column(Text, nullable=True)

    last_heartbeat_at = Column(DateTime, nullable=True)

    created_at = Column(DateTime, default=dt.datetime.utcnow)
    updated_at = Column(DateTime, default=dt.datetime.utcnow, onupdate=dt.datetime.utcnow)

    store = relationship("Store", back_populates="cameras")
    shelves = relationship("Shelf", back_populates="camera")

File: backend/models/camera.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/db');

const Camera = sequelize.define('Camera', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  camera_id: {
    type: DataTypes.STRING(50),
    allowNull: false,
    unique: true
  },
  store_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'stores',
      key: 'id'
    }
  },
  name: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  camera_type: {
    type: DataTypes.ENUM('fixed', 'ptz', 'dome', 'bullet'),
    defaultValue: 'fixed'
  },
  camera_url: {
    type: DataTypes.STRING(500)
  },
  stream_url: {
    type: DataTypes.STRING(500)
  },
  position_x: {
    type: DataTypes.FLOAT
  },
  position_y: {
    type: DataTypes.FLOAT
  },
  height: {
    type: DataTypes.FLOAT
  },
  angle: {
    type: DataTypes.FLOAT
  },
  is_active: {
    type: DataTypes.BOOLEAN,
    defaultValue: true
  },
  fps: {
    type: DataTypes.INTEGER,
    defaultValue: 5
  },
  resolution: {
    type: DataTypes.STRING(20),
    defaultValue: '640x480'
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  },
  updated_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'cameras',
  timestamps: false
});

module.exports = Camera;